lib/impress.application.js

```diff
@@ -575,6 +586,96 @@ impress.application.mixin = (application) => {
     }
   };
 
+  if (impress.mode === 'test') {
+
+    application.loadTests = (callback) => {
+      const relPath = '/test/';
+      const absPath = application.dir + relPath;
+      api.fs.readdir(absPath, (err, files) => {
+        if (err) {
+          application.log.error(impress.CANT_READ_DIR + absPath);
+          callback();
+          return;
+        }
+        if (files.length === 0) {
+          callback(null);
+          return;
+        }
+        application.tests = [];
+        api.metasync.each(files, (fileName, cb) => {
+          const filePath = absPath + fileName;
+          api.fs.stat(filePath, (err, stats) => {
+            if (err) {
+              application.log.error(impress.CANT_READ_FILE + filePath);
+              cb();
+              return;
+            }
+            if (!stats.isFile()) {
+              cb();
+              return;
+            }
+            application.loadTest(fileName, filePath, cb);
+          });
+        }, callback);
+      });
+    };
+
+    application.loadTest = (fileName, path, callback) => {
+      const fileExt = api.common.fileExt(fileName);
+      if (fileExt !== 'js') {
+        callback();
+        return;
+      }
+      const testName = api.common.removeExt(fileName);
+      application.createTest(path, (err, test) => {
+        if (err) {
+          callback(err);
+          return;
+        }
+        application.tests[testName] = test;
+        callback();
+      });
+    };
+
+    application.createTest = (fileName, callback) => {
+      api.fs.readFile(fileName, (err, source) => {
+        if (err) {
+          application.log.error(impress.CANT_READ_FILE + fileName);
+          callback(err);
+        } else {
+          const options = {
```

Comment:
Since `USE_STRICT` is equal to `'\'use strict\';\n'`, and `testWrap` function puts it before the code, I'd recommend to add `lineOffset` of 1 to options for more accurate results.